src/module/skb_drop/skb_drop.rs

```diff
@@ -63,13 +68,23 @@ impl Collector for SkbDropModule {
     }
 
     fn init(&mut self, _: &CliConfig, probes: &mut ProbeManager) -> Result<()> {
-        let symbol = Symbol::from_name("skb:kfree_skb")?;
+        let mut probe = Probe::raw_tracepoint(Symbol::from_name("skb:kfree_skb")?)?;
+        let hook = Hook::from(skb_drop_hook::DATA);
 
-        if let Err(e) = probes.register_probe(Probe::raw_tracepoint(symbol)?) {
+        if self.reasons_available {
+            probes.register_kernel_hook(hook)?;
```

Comment:
I'm wondering what other functions (apart from `kfree_skb_reason`) are we interested in probing?
If a kernel module does some internal processing where it uses core's `enum skb_drop_reason` , do we want to probe it?

Maybe it would be just easier to add a directed hook here as well?

Reply:
I'm not suggesting we hook onto more probes. On the contrary, I'm asking why do we need to register the kernel hook for user-provided hooks. I just don't see the case where it'll be needed/wanted.

Reply:
Right. I don't see much value and it can even be confusing if some module manipulates error codes without them being actually applied to kfree_skb_reason.

Reply:
Yeah, I meant to say `-p dropmon collect -p...` or the  similar `-c skb,skb-drop -p...`
My general understanding is that skb-drop is all about skb drops, not any function that handles `skb_drop_reason`.
I just don't see a use-case for extracting that information from anything else but that one tracepoint. So we're adding that hook to every single probe and taking one out of the 10 slots we have without providing any real value.

Reply:
> Maybe the correct answer here would be to have two collectors: skb-drop and drop-reason and to let the user combine them (and of course the dropmon profile in the first place). Or to simply remove the probe on tp:skb:kfree_skb and let that decision to the profile and/or user?

Ok, ok. I've failed to convince you to remove the generic hook, but let's not do the exact opposite xD